zebra-chain/src/transaction/serialize.rs

```diff
@@ -321,6 +344,40 @@ impl ZcashDeserialize for Transaction {
                 let inputs = Vec::zcash_deserialize(&mut reader)?;
                 let outputs = Vec::zcash_deserialize(&mut reader)?;
 
+                let value_balance = (&mut reader).zcash_deserialize_into()?;
+                let shared_anchor = sapling::tree::Root(reader.read_32_bytes()?);
+                let mut shielded_spends = Vec::zcash_deserialize(&mut reader)?;
+                let mut shielded_outputs = Vec::zcash_deserialize(&mut reader)?;
+
+                // We can deserialize all sapling data in one go in V5 as the
+                //  internal structure line up with the serialized structure.
```

Comment:
In the spec, the spends are split up into 3 arrays, and the outputs into 2 arrays.

So we can't serialise or deserialise them as a vector of spends or outputs, we have to deserialise their components, and then create the spends or outputs from their parts.

![02C37C54-15F1-43ED-9414-87B426E311BC](https://user-images.githubusercontent.com/8951843/114324440-4e294480-9b6d-11eb-8a9f-dfdfe93b98c4.jpeg)